### PR TITLE
meta: write stubs for command-chain using hooks when needed

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: tests/unit/meta/test_meta.py [hunk 1]
```diff
@@ -1110,6 +1110,48 @@ def test_snap_hooks_overwrite_part_hooks(self):
...
         )
 
+    def test_hooks_with_command_chain_write_stub(self):
+        self.config_data["hooks"] = {
+            "configure": {"command-chain": ["fake-hook-command-chain"]}
+        }
+
+        self.generate_meta_yaml()
+
+        self.assertThat(os.path.join(self.hooks_dir, "configure"), FileExists())
```
Comment (cjp256) [inline tests/unit/meta/test_meta.py:1120]: This matches surrounding code, but i thought we were moving away from testtools...? Should new tests be written in it?
something like:
```
configure_hook = pathlib.Path(self.hooks_dir, "config")

assert configure_hook.read_text() == "#!/bin/sh\n"
```

OK either way, just trying to be consistent :+1:

File: tests/unit/meta/test_meta.py [hunk 2]
```diff
@@ -1110,6 +1110,48 @@ def test_snap_hooks_overwrite_part_hooks(self):
...
+        # Setup a prime/snap directory containing a hook.
+        part_hook = os.path.join(self.prime_dir, "snap", "hooks", "install")
+        _create_file(part_hook, content="from part", executable=True)
+
+        # Now write the snap directory, and verify that the snap hook overwrote
+        # the part hook in both prime/snap/hooks and prime/meta/hooks.
+        self.generate_meta_yaml()
+
+        self.assertThat(
+            os.path.join(self.hooks_dir, "install"),
```
Comment (cjp256) [inline tests/unit/meta/test_meta.py:1142]: Should we verify the snap/hooks/intall is expected?

**Review by cjp256 — COMMENTED** [review]
Hooks are complicated. II'll need some :coffee: in the morning and will do a fresh pass :grinning:

File: tests/unit/meta/test_meta.py (hunk 1 above)
Comment (sergiusens) [reply tests/unit/meta/test_meta.py:1120]: consistency, yes

File: tests/unit/meta/test_meta.py (hunk 2 above)
Comment (sergiusens) [reply tests/unit/meta/test_meta.py:1142]: there's already a test for that above or below :-)

Comment (sergiusens) [reply tests/unit/meta/test_meta.py:1142]: on a second look, snap/hooks/install is what I setup in the test setup, is the expactation to always test that what was setup still exists after excercising anything?

Comment (cjp256) [reply tests/unit/meta/test_meta.py:1142]: I was just thinking of validating that this logic didn't break the snap/hooks.  But looking back at the code, seems unnecessary.

**Review by cjp256 — APPROVED** [review]

**Pull request merged**